README.md

```diff
@@ -1,107 +1,148 @@
 
-Contracts can use Storage to persist data between calls. The simplest way to read and write data into Storage is to add fields to the Contract struct:
-```rust
-#[contract]
-struct PrinceTheDog {
-    age: u8, 
-    breed: String,
-    hungry: bool,
-    toy: DogToy,
-}
-```
+Note that having a `&self` receiver or no receiver at all only prevents mutations from being done from the receiver, it doesn't prevent mutations from being done generally. So for example, a method with a `&self` receiver could still mutate the world state by calling `pchain_sdk::storage::set`.
 
-The `#[contract]` macro transparently generates code that loads all Contract struct's fields from Storage before the execution of contract methods. All types that implement the `Storage` trait can be used as a Contract field. Out of the box, this includes all Rust primitive types, as well as other commonly used types like `Option<T>`, `Result<T>`, `Vec<T>`, etc. In addition, structs defined by the developer can be made to implement `Storage` by applying the `#[contract_field]` macro on their definitions, if all *their* fields implement Storage:
-```rust
-#[contract_field]
-struct DogToy {
-    ...
-}
-```
+## More about Contract Storage
 
-### Storage and Collections
+### Cacher
 
-Because Storage is so gas-expensive, loading all Contract's fields before Method execution and writing them all into Storage after execution typically results in Contracts that are not very economical. For Contracts that do not keep much in Storage, or whose Methods *always* read and write into most fields, this may be okay, or even ideal, however, some applications cannot avoid keeping a lot of on-chain state, and for these applications eagerly loading and saving fields in every call may be unacceptably expensive.
+As explained previously, by default, the SDK loads *all* of a contract's fields from storage before executing a method, and saves all of them into storage after the execution completes.
 
-To solve this, the SDK includes a `pchain_sdk::collections` module. All of the types defined in this module 'lazily' load Storage: they only incur a read or write gas cost when the exact item in the collection is read from or written to. They also offer an API that can make working with large collections of data more convenient.
+This default could be ideal for contracts that do not keep much in storage, or whose methods *always* read and write into every field in every call, but will in general result in contract calls that are not very gas-efficient.
 
 ```rust
 #[contract]
 struct PrinceTheDog {
-    nicknames: Vector<String>
+    // `Cacher` can wrap around any type that implements `Storable`.
+    age: Cacher<u8>,
+    breed: Cacher<String>,
+    hungry: Cacher<bool>,
+    toy: Cacher<DogToy> 
```

Comment:
@fabricwave this example wraps `Cacher` directly around `DogToy`.

Questions: 
1. What happens when we instead wrap `Cacher` around `DogToy`'s fields individually?
2. What happens when we both wrap around `DogToy` (like currently), and also wrap around `DogToy`'s fields individually?